### PR TITLE
Display error message when language file is missing

### LINKED ISSUE #31384: BO - Translations - 2 Exceptions displayed when trying to add/update language Breton
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

2 Exceptions are displayed when I add/update the language Breton.

**Add/update language Breton:**
`Invalid or uninitialized Zip object`
Screenshot 2023-02-15 at 15 04 49

**Refresh your URL:**
`An exception has been thrown during the rendering of a template ("Parameter "type" for route "api_translation_domains_tree" must match "[^/]++" ("" given) to generate a corresponding URL.").`
Screenshot 2023-02-15 at 11 30 51

https://user-images.githubusercontent.com/16019289/219048868-d764da40-3d03-42ef-b2be-734509b907eb.mov

It's a regression. In 1.7.8.8, we could add/update Breton language without error.

### Expected behavior

Display an error message when the language translation is not available

### Steps to reproduce

1. Go to BO > International > Translations
2. Add language Breton
3. See error :x: `Invalid or uninitialized Zip object`
4. Update your URL
5. See Error :x: `An exception has been thrown during the rendering of a template ("Parameter "type" for route "api_translation_domains_tree" must match "[^/]++" ("" given) to generate a corresponding URL.").`
6. Refresh and do that multiple times
7. Internal Server Error :x:

### PrestaShop version(s) where the bug happened

8.1.0

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by Prestaworks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.0.x
| Description?      | If language file is missing, display error message.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Follow instructions in issue
| Fixed ticket?     | Fixes #31384
| Sponsor company   | Prestaworks AB

### CODE REVIEW

**Comment by MatShir** [comment]
Thank for the fix ! 
I need to check why in the first place there is no zip.
To investigate with @l-delin

File: classes/Language.php [hunk 1]
```diff
@@ -1263,6 +1263,13 @@ public static function downloadXLFLanguagePack($locale, &$errors = [], $type = s
             return false;
         }
 
+        //Check if response is a zip file
+        if (strpos($content, "\x50\x4b\x03\x04") === false) {
+            $errors[] = Context::getContext()->getTranslator()->trans('Language pack unavailable.', [], 'Admin.International.Notification') . ' ' . $url;
```
Comment (l-delin) [inline classes/Language.php:1262]: Hello, are we talking about localization packs or something else? Thanks!

Comment (Prestaworks) [reply classes/Language.php:1262]: Hi,

This file is failing
https://i18n.prestashop-project.org/translations/8.0.1/br-FR/br-FR.zip
thats a translation pack I think (xlf files).

Comment (l-delin) [reply classes/Language.php:1262]: Alright :-) the wording is all good to go, thank you!

**Review by kpodemski — DISMISSED** [review]

**Review by 0x346e3730 — DISMISSED** [review]

**Review by SharakPL — CHANGES_REQUESTED** [review]
https://github.com/PrestaShop/PrestaShop/blob/c6767d0edd687f4c7c887fee42dc60b9ee269321/classes/Language.php#L1260-L1271

No need for this duplication.

Suggestion:
```php
 // strpos checks if response is a zip file 
 if (empty($content) || strpos($content, "\x50\x4b\x03\x04") === false) { 
     $errors[] = Context::getContext()->getTranslator()->trans('Language pack unavailable.', [], 'Admin.International.Notification') . ' ' . $url; 
  
     return false; 
 }
```

**Review by kpodemski — COMMENTED** [review]
@Prestaworks oh, @SharakPL is right, I think we could use the above's code block

**Review by florine2623 — COMMENTED** [review]
Hello @Prestaworks ,

Thanks for the PR !

I do have an error message when the language pack doesn't exist :
Screenshot 2023-03-31 at 13 39 37

https://user-images.githubusercontent.com/16019289/229110695-1900d7bd-afc6-4644-90ec-f417a66d7b29.mov


Although the error message, to me, seems a little confusing. I want to click on the link, but just an XML is displayed.
Screenshot 2023-03-31 at 13 40 30

Maybe the message could be improved ?
@l-delin could you give your insight on this please ? :)

Thanks !

**Comment by l-delin** [comment]
Is there a way to display the error directly in the error message? 

For example:
- The language pack is not available. **<Message>The specified key does not exist.</Message>** Then, a dropdown button that can be clicked showing the details of the error:
**<Details>No such object: translations-core-oss-production/translations/8/br-FR/br-FR.zip</Details>**

As in:
image

image

Ping @PrestaShop/ux-team, what do you think?

**Comment by ElodieOS** [comment]
Hi @l-delin ! 
I don't think it's necessary to hide the information in a dropdown list, as it's not very big. 
However, in terms of UX, can we have a solution to solve this problem? Because here, we understand that there is a problem but we don't have a solution to solve it. If we have a small paragraph to explain which action do, it will be better and the dropdown will make sens to keep the alert banner at the right size. 

Wdyt?

**Comment by Prestaworks** [comment]
How about merging the PR so the issue is fixed and then open a new Ticket with suggestions and discussions for improving the display of error messages on the prestashop-projects site?

**Comment by MatShir** [comment]
> How about merging the PR so the issue is fixed and then open a new Ticket with suggestions and discussions for improving the display of error messages on the prestashop-projects site?

Sounds good to me 👍

**Comment by MatShir** [comment]
@florine2623 is the PR QA approved?

**Comment by florine2623** [comment]
@MatShir ,

It is QA ✅ 

Ok for taking it further in discussions ^^

**Comment by MatShir** [comment]
Oh I didn't see @SharakPL change request. @Prestaworks could you apply them? We will test your fix straight away

**Review by nicosomb — APPROVED** [review]

**Review by SharakPL — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Thanks @Prestaworks ,

Works as expected. The error message is displayed when no language exists.
Add/Update language for other language that Breton works as usual.

It is QA ✅

**Review by boherm — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
thanks Daniel @Prestaworks :)

**Comment by l-delin** [comment]
Hello, two things:

1. Is this a patch?

2. I don't think the error message as it's been merged is intuitive at all for users.

As @florine2623  said, "Although the error message, to me, seems a little confusing. I want to click on the link, but just an XML is displayed." I believe this needs to be fixed.

I opened an issue regarding this topic: https://github.com/PrestaShop/PrestaShop/issues/32126

Thanks

**Comment by MatShir** [comment]
It's for a patch version 8.0.4. If we want to improve the interface, we can address the issue for the next major/minor release

**Pull request merged**